lib/modules/manager/gomod/artifacts.ts

```diff
@@ -296,7 +299,22 @@ export async function updateArtifacts({
 
     const execCommands: string[] = [];
 
-    let args = 'get -d -t ./...';
+    let goGetDirs: string | undefined;
+    if (config.goGetDirs) {
+      goGetDirs = config.goGetDirs
+        .filter((dir) => {
+          const isValid = isValidLocalPath(dir);
+          if (!isValid) {
+            logger.warn({ dir }, 'Invalid path in goGetDirs');
+          }
+          return isValid;
+        })
```

Comment:
please extract a helper function

Reply:
That is really a matter of taste, why extract this to a function that is not used anywhere else that pollutes the package?

what would you even call that function? `isValidLocalPathAndLog` ?

To be honest that's the kind of change that as a maintainer you can do after this is merged.